tests/frontend/detector_descriptor/test_detector_descriptor_base.py

```diff
@@ -74,6 +74,32 @@ def test_computation_graph(self):
                 self.assertEqual(keypoints, expected_kps)
                 np.testing.assert_array_equal(descriptors, expected_descs)
 
+    def test_filter_by_mask(self) -> None:
+        """Test the `filter_by_mask` method."""
+        # Create a (9, 9) mask with ones in a (5, 5) square in the center of the mask and zeros everywhere else.
+        mask = np.zeros((9, 9)).astype(np.uint8)
+        mask[2:7, 2:7] = 1
+
+        # Test coordinates near corners of square of ones and along the diagonal.
+        coordinates = np.array(
+            [
+                [1.4, 1.4],
+                [1.4, 6.4],
```

Comment:
any reason to have the subpixel coordinates? i think makes it a bit less readable. but if there's a specific use case you're targeting, maybe that's ok.

I don't think any of our current detectors provide subpixel resolution.

Reply:
Most feature keypoints are subpixel (including SIFT), as detections are usually computed by aggregating information at multiple scales.